File: src/main/webapp/src/app/models/midi-device.ts
export class MidiDevice {
    id: number = 0;
    name: string = '';
    vendor: string = '';
    description: string = '';
    serialPort: boolean = false;

    constructor(data?: any) {
        if(!data) {
        	return;
        }
        this.id = data.id;
        this.name = data.name;
        this.vendor = data.vendor;
        this.description = data.description;
        this.serialPort = data.serialPort;
    }
}

File: src/main/java/com/ascargon/rocketshow/midi/MidiDevice.java
package com.ascargon.rocketshow.midi;

import lombok.Getter;
import lombok.Setter;

/**
 * A MIDI device containing name and id.
 *
 * @author Moritz A. Vieli
 */
@Getter
@Setter
public class MidiDevice {

    private int id;
    private String name;
    private String vendor;
    private String description;
    private boolean serialPort = false;

}
